filament/src/details/ColorGrading.cpp

```diff
@@ -688,170 +691,206 @@ FColorGrading::FColorGrading(FEngine& engine, const Builder& builder) {
 
-                        // Saturation in linear space
-                        v = saturation(v, c.colorGradingLuminance, builder->saturation);
+            // Kill negative values before curves
+            v = max(v, 0.0f);
 
-                        // Kill negative values before curves
-                        v = max(v, 0.0f);
+            // RGB curves
+            v = curves(v,
+                    builder->shadowGamma, builder->midPoint, builder->highlightScale);
+        }
 
-                        // RGB curves
-                        v = curves(v,
-                                builder->shadowGamma, builder->midPoint, builder->highlightScale);
-                    }
+        // Tone mapping
+        if (builder->luminanceScaling) {
+            v = luminanceScaling(v, *builder->toneMapper, config.colorGradingLuminance);
+        } else {
+            v = (*builder->toneMapper)(v);
+        }
 
-                    // Tone mapping
-                    if (builder->luminanceScaling) {
-                        v = luminanceScaling(v, *builder->toneMapper, c.colorGradingLuminance);
-                    } else {
-                        v = (*builder->toneMapper)(v);
-                    }
+        // Go back to display color space
+        v = config.colorGradingOut * v;
 
-                    // Go back to display color space
-                    v = c.colorGradingOut * v;
+        // Apply gamut mapping
+        if (builder->gamutMapping) {
+            // TODO: This should depend on the output color space
+            v = gamutMapping_sRGB(v);
+        }
 
-                    // Apply gamut mapping
-                    if (builder->gamutMapping) {
-                        // TODO: This should depend on the output color space
-                        v = gamutMapping_sRGB(v);
-                    }
+        // TODO: We should convert to the output color space if we use a working
+        //       color space that's not sRGB
+        // TODO: Allow the user to customize the output color space
 
-                    // TODO: We should convert to the output color space if we use a working
-                    //       color space that's not sRGB
-                    // TODO: Allow the user to customize the output color space
+        // We need to clamp for the output transfer function
+        v = saturate(v);
 
-                    // We need to clamp for the output transfer function
-                    v = saturate(v);
+        // Apply OETF
+        v = config.oetf(v);
 
-                    // Apply OETF
-                    v = c.oetf(v);
+        return v;
+    };
 
-                    *p++ = half4{v, 0.0f};
-                }
+    //auto now = std::chrono::steady_clock::now();
+
+    if (mIsOneDimensional) {
+        uint8_t* UTILS_RESTRICT p = (uint8_t*) data;
+        for (size_t rgb = 0; rgb < config.lutDimension; rgb++) {
+            float s;
+            if (mIsLDR) {
```

Comment:
move this out of the loop, or even just do two loops?

Reply:
I was thinking the same thing after I went home for the day